spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/Recurrent.scala

```diff
@@ -219,13 +220,45 @@ class Recurrent[T : ClassTag]()
     // init state
     currentInput(hidDim) = if (initState != null) initState
      else hidden
+
+    if (feedbackOutput) {
+      newInput = Tensor(input.size())
+    }
+
     while (i <= times) {
-      currentInput(inputDim) = outputCell.select(timeDim, i)
+      if (!feedbackOutput) {
+        currentInput(inputDim) = outputCell.select(timeDim, i)
+      } else {
+        val inputTmp = if (i == 1) {
+          // input at t(0) is last time step of user input
+          input.select(timeDim, times).clone()
+        } else {
+          // input at t(i) is output at t(i-1)
+          cells(i - 2).output.toTable[Tensor[T]](inputDim)
+        }
+        newInput.narrow(timeDim, i, 1).copy(inputTmp)
+        require(inputTmp.nElement() == input.select(timeDim, i).nElement(), "outputsize is " +
+          "not the same with input size!! Please update cell settings or turn off feedbackOutput.")
+        currentInput(inputDim) = if (preTopology != null) {
+          val sizes = 1 +: inputTmp.size()
+          inputTmp.resize(sizes)
+          val _input = preTopology.updateOutput(inputTmp).toTensor[T]
+          inputTmp.resize(sizes.takeRight(sizes.length - 1))
+          _input.select(1, 1)
```

Comment:
I think the performance is poor using this setting. How about modify Cell layer to let preTopology be a choice.
Then if feedbackOutput == true, simply define another while loop.